sky/clouds/azure.py

```diff
@@ -137,6 +134,43 @@ def get_egress_cost(self, num_gigabytes: float):
     def is_same_cloud(self, other):
         return isinstance(other, Azure)
 
+    @classmethod
+    def get_image_size(cls, image_id: str, region: Optional[str]) -> float:
+        if region is None:
+            # The region used here is only for where to send the query,
+            # not the image location. Azure's image is globally available.
+            region = 'eastus'
+        if image_id.startswith('skypilot:'):
+            image_id = service_catalog.get_image_id_from_tag(image_id,
+                                                             clouds='azure')
```

Comment:
For the tagged images, let's shortcircuit the size check by directly setting the image size to the number we know. This is to optimize the speed for creating `Resources` object in most cases.

Reply:
Good point! Added.